src/Chapter3.hs

```diff
@@ -480,6 +511,47 @@ After defining the city, implement the following functions:
    and at least 10 living __people__ inside in all houses of the city in total.
 -}
 
+data City = City {
+  castle :: Maybe (Castle, Maybe Wall),
```

Comment:
That's a correct representation! 👍🏻 Though, I feel like it's a bit awkward to work with this type. You can create a custom data type to model the same structure:

```haskell
data Castle
    = NoCastle
    | Castle String
    | CastleWithWalls String
```

Creating custom types in Haskell is cheap, so no harm in introducing custom domain data types 🙂